service/mantle/training/TrainingServices.xml

```diff
@@ -0,0 +1,136 @@
+            <parameter name="price" required="true"/>
+            <parameter name="description"/>
+            <parameter name="priceUomId" default-value="USD"/>
+        </in-parameters>
+        <out-parameters>
+            <parameter name="productId"/>
+
+        </out-parameters>
+        <actions>
+            <service-call name="create#mantle.product.Product" in-map="context" out-map="context"/>
+
+            <service-call name="create#mantle.product.ProductPrice" in-map="context + [productId:productId]"
+                          out-map="context"/>
+        </actions>
+    </service>
+
+
+     <service verb="get" noun="ProductOne">
+        <description>To get the details of a specific</description>
+        <in-parameters>
+            <parameter name="productId" required="true"/>
+        </in-parameters>
+        <out-parameters>
+            <parameter name="productId"/>
+            <parameter name="description"/>
+            <parameter name="productName"/>
+            <parameter name="price"/>
+        </out-parameters>
+        <actions>
+            <entity-find-one entity-name="mantle.product.Product" value-field="product"/>
+            <set field="productId" from="product.productId"/>
+            <set field="description" from="product.description"/>
+            <set field="productName" from="product.productName"/>
+            <entity-find-one entity-name="mantle.product.ProductPrice" value-field="productPrice"/>
+            <set field="price" from="productPrice.price"/>
+            <set field="priceUomId" from="productPrice.priceUomId"/>
+        </actions>
+    </service>
+
+
+    <service verb="get" noun="Product">
+        <description>To get details of all the products</description>
+
+        <out-parameters>
+            <parameter name="products"/>
+        </out-parameters>
+        <actions>
+
+            <entity-find entity-name="Product" list="productDetailList">
+                <select-field field-name="productId"/>
+                <select-field field-name="productName"/>
+                <select-field field-name="description"/>
+            </entity-find>
+            <entity-find entity-name="ProductPrice" list="productPriceList">
+                <select-field field-name="productId"/>
+                <select-field field-name="price"/>
+                <select-field field-name="priceUomId"/>
+            </entity-find>
+            <set field="productsList" from="[]"/>
+            <iterate list="productPriceList" entry="ProductPrices">
+                <entity-find-one entity-name="mantle.product.Product" value-field="Products">
+                    <field-map field-name="productId" from="ProductPrices.productId"/>
+                </entity-find-one>
+
+                <set field="tempMap" from="[productId:Products.productId,productName:Products.productName,description:Products.description,price:ProductPrices.price,priceUomId:ProductPrices.priceUomId]
```

Comment:
We can use names like productMap. names like tempMap is not the right way, we should always have meaningful names for our variables/services indicating the purpose of each.